docker_test.go

```diff
@@ -795,4 +883,972 @@ func consume(topics []string, master sarama.Consumer) (chan *sarama.ConsumerMess
+										END LOOP;
+								END ;
+								$$ LANGUAGE plpgsql`
+	config.Load()
+	logger.Init()
+	postgres.Init()
+	mssql.Init()
+	clickhouse.Init()
+}
+
+// SetWhPostgresDestination setup warehouse postgres destination
+func SetWhPostgresDestination(pool *dockertest.Pool) (output func()) {
+	pgTest := whTest.pgTest
+	credentials := pgTest.credentials
+	output = func() {}
+
+	var err error
+	if pgTest.resource, err = pool.Run("postgres", "11-alpine", []string{
+		fmt.Sprintf("POSTGRES_DB=%s", credentials.DbName),
+		fmt.Sprintf("POSTGRES_PASSWORD=%s", credentials.Password),
+		fmt.Sprintf("POSTGRES_USER=%s", credentials.User),
+	}); err != nil {
+		panic(fmt.Errorf("Could not create WareHouse Postgres: %v\n", err))
+	}
+
+	// Getting at which port the postgres container is running
+	credentials.Port = pgTest.resource.GetPort("5432/tcp")
+
+	purgeResources := func() {
+		if pgTest.resource != nil {
+			log.Printf("Purging warehouse postgres resource: %s \n", err)
+			if err := pool.Purge(pgTest.resource); err != nil {
+				log.Printf("Could not purge warehouse postgres resource: %s \n", err)
+			}
+		}
+	}
+
+	if err = pool.Retry(func() error {
+		var err error
+		pgTest.db, err = postgres.Connect(*credentials)
+		if err != nil {
+			return err
+		}
+		return pgTest.db.Ping()
+	}); err != nil {
+		defer purgeResources()
+		panic(fmt.Errorf("Could not connect to warehouse postgres with error: %w\n", err))
+	}
+	output = purgeResources
+	return
+}
+
+// SetWhClickHouseDestination setup warehouse clickhouse destination
+func SetWhClickHouseDestination(pool *dockertest.Pool) (output func()) {
+	chTest := whTest.chTest
+	credentials := chTest.credentials
+	output = func() {}
+
+	var err error
+	if chTest.resource, err = pool.Run("yandex/clickhouse-server", "21-alpine", []string{
+		fmt.Sprintf("CLICKHOUSE_DB=%s", credentials.DbName),
+		fmt.Sprintf("CLICKHOUSE_PASSWORD=%s", credentials.Password),
+		fmt.Sprintf("CLICKHOUSE_USER=%s", credentials.User),
+	}); err != nil {
+		panic(fmt.Errorf("Could not create WareHouse ClickHouse: %v\n", err))
+	}
+
+	// Getting at which port the clickhouse container is running
+	credentials.Port = chTest.resource.GetPort("9000/tcp")
+
+	purgeResources := func() {
+		if chTest.resource != nil {
+			log.Printf("Purging warehouse clickhouse resource: %s \n", err)
+			if err := pool.Purge(chTest.resource); err != nil {
+				log.Printf("Could not purge warehouse clickhouse resource: %s \n", err)
+			}
+		}
+	}
+
+	if err = pool.Retry(func() error {
+		var err error
+		chTest.db, err = clickhouse.Connect(*credentials, true)
+		if err != nil {
+			return err
+		}
+		return chTest.db.Ping()
+	}); err != nil {
+		defer output()
+		panic(fmt.Errorf("Could not connect to warehouse clickhouse with error: %w\n", err))
+	}
+	output = purgeResources
+	return
+}
+
+// SetWhClickHouseClusterDestination setup warehouse clickhouse cluster mode destination
+func SetWhClickHouseClusterDestination(pool *dockertest.Pool) (output func()) {
```

Comment:
```suggestion
func SetWHClickHouseClusterDestination(pool *dockertest.Pool) (cleanup func()) {
```

1. `output` is a bit confusing. Also go naming convention for initial is that all letters need to be either upper-case or lower-case.

2. So `WH` instead of `Wh`, `customerID` instead of `customerId`, `customerUUID` instead of `customerUuid`

3. What if this method also defined and returned: 

```
ClickHouseCusterTest{
		writeKey: randString(27),
		credentials: &clickhouse.CredentialsT{
			Host:          "localhost",
			User:          "rudder",
			Password:      "rudder-password",
			DbName:        "rudderdb",
			Secure:        "false",
			SkipVerify:    "true",
			TlsConfigName: "",
		},
	}
```
instead of happening in  initWhConfig().

This way would be more compact and the code can be live stand alone in a library, so it can be used in other places as well.

Reply:
(The above apply to all similar functions.

Reply:
Moved destination setup config in their respective setup's. Also changes naming convention as well.

Reply:
Done.